sky/serve/replica_managers.py

```diff
@@ -918,3 +1002,28 @@ def _replica_prober(self) -> None:
                 with ux_utils.enable_traceback():
                     logger.error(f'  Traceback: {traceback.format_exc()}')
             time.sleep(serve_constants.ENDPOINT_PROBE_INTERVAL_SECONDS)
+
+    #########################################
+    # SkyServe Update and replica versioning. #
+    #########################################
+
+    def update_version(self, version: int, spec: 'service_spec.SkyServiceSpec',
+                       mixed_replica_versions: bool,
+                       task_yaml_path: str) -> None:
+        self.version2spec[version] = spec
+        self.version2yaml[version] = task_yaml_path
+        self.latest_version = version
+        self.mixed_replica_versions = mixed_replica_versions
+        self._task_yaml_path = task_yaml_path
+
+    def get_latest_version(self) -> int:
+        return self.latest_version
+
+    def _get_readiness_path(self, version: int) -> str:
+        return self.version2spec[version].readiness_path
+
+    def _get_post_data(self, version: int) -> Optional[Dict[str, Any]]:
+        return self.version2spec[version].post_data
+
+    def _get_initial_delay_seconds(self, version: int) -> int:
+        return self.version2spec[version].initial_delay_seconds
```

Comment:
nit: Could we make those functions into attributes?

Reply:
I think if it takes `version` as an input, it is probably better to get them as getter functions?

Reply:
Oh nvm. I misread it. Lets keep current impl